### ISSUE TITLE
Hasura CE Open Source limitations

### ISSUE THREAD

**Issue by Hongbo-Miao** [Hongbo-Miao/hongbomiao.com#4782]
https://github.com/hasura/graphql-engine

Limitations:

- No rate limits and query depth limits: https://github.com/hasura/graphql-engine/issues/2151
  - For query depth limits, better option: allow list https://hasura.io/docs/latest/graphql/core/deployment/allow-list/
  <img width="910" alt="image" src="https://user-images.githubusercontent.com/3375461/165588792-7cb8c176-0822-43e2-b54f-759871ea204c.png">
- No Prometheus: https://github.com/hasura/graphql-engine/issues/6336
  - Community solutions:
    - hasura-metric-adapter : https://github.com/afitzek/hasura-metric-adapter 
      - Including hasura-exporter features: https://github.com/afitzek/hasura-metric-adapter/pull/4
    - hasura-exporter: https://github.com/zolamk/hasura-exporter

**Comment by Hongbo-Miao** [comment]
Allow list added by https://github.com/Hongbo-Miao/hongbomiao.com/pull/4795
hasura-metric-adapter added by https://github.com/Hongbo-Miao/hongbomiao.com/pull/4802

**Comment by edulix** [comment]
What about running multiple instances of Hasura for scaling and High Availability? According to the [documentation](https://hasura.io/learn/graphql/hasura-advanced/reliability/4-high-availability/):

> "Also, with multiple instances of Hasura connected to the same database, you might run into issues such as the event triggers duplicating events or unreliable subscriptions. Hasura HA eliminates those issues."

The unreliable subscriptions probably can be fixed using [Traefik sticky sessions](https://traefik.io/glossary/what-are-sticky-sessions/). 

For solving the duplication of events the only idea I have is maintaining a single node with the event configuration but that seems messy. Any ideas?

**Comment by hongbo-miao** [comment]
Hi @edulix sorry I muted this repo.. The bot keeps notifying me. Just saw the message. But the new custom notification feature allows me to select what notifications I can receive for this repo. Hopefully not missed any more in future..

Also feel free to use [discussions](https://github.com/hongbo-miao/hongbomiao.com/discussions) too!

Hmm I think personally I would avoid using Hasura events and use Postgres CDC through Debezium for example instead.
Even Hasura supports SQL migrations and events, but there are more professional tools there. If Hasura does all those things, it will be too risky and hard to scale. Hopefully it makes sense! ☺️

**Comment by hongbo-miao** [comment]
Hey @edulix, just noticed your message. I had muted this repo due to constant notifications from the bot about pull requests. However, I've now updated my GitHub CODEOWNERS file, so everything should be good. Repo is unmuted!

Regarding Hasura events, I'd personally steer clear and opt for Postgres CDC with Debezium instead. While Hasura does support things like events and SQL migrations, using specialized tools for these tasks seems more professional. Relying solely on Hasura for everything could be risky and challenging to scale. Hope that makes sense! ☺️
